baseapp/baseapp.go

```diff
@@ -701,6 +701,11 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 		return
 	}
 
+	var startingGas uint64
+	if mode == runTxModeDeliver {
```

Comment:
Why do we need a check for DeliverTx?

Reply:
Otherwise we might not have `ctx.BlockGasMeter()`